src/controllers/users.js

```diff
@@ -117,29 +106,88 @@ export default class {
    * @param {function} next
    * @returns {object} responses
    */
-  static async resendEmail(req, res, next) {
-    try {
-      const { email } = req.query;
-      const hostUrl = `${req.protocol}://${req.get('host')}`;
-      const user = await db.user.findOne({
-        where: { email }
+  async resendEmail(req, res) {
+    const { email } = req.query;
+    const host = `${req.protocol}://${req.get('host')}`;
+    const user = await db.user.findOne({
+      where: { email }
+    });
+    if (user) {
+      const token = await JWTHelper.signToken(user);
+      const mail = new Mailer({
+        name: user.firstName,
+        to: user.email,
+        token,
+        host
       });
-      if (user) {
-        const token = await JWTHelper.signToken(user);
-        await Mailer.sendVerificationEmail(user.firstName, hostUrl, user.email, token);
-        return Responses.handleSuccess(
-          200,
-          'Email has been resent successfully, please check your mail',
-          res
-        );
-      }
-      return Responses.handleError(
-        404,
-        'No account with such email exists, please sign up',
+      await mail.sendVerificationEmail();
+      return Responses.handleSuccess(
+        200,
+        'Email has been resent successfully, please check your mail',
         res
       );
-    } catch (err) {
-      next(new ErrorHandler('Error at verification'));
     }
+    return Responses.handleError(
+      404,
+      'No account with such email exists, please sign up',
+      res
+    );
+  }
+
+  /**
+   * forgot password controller
+   * @param {*} req
+   * @param {*} res
+   * @returns {*} responses
+   */
+  async forgotPassword(req, res) {
+    const host = `${req.protocol}://${req.get('host')}`;
+    const user = await UserServices.findUserByEmail(req.body.email);
+    if (!user) {
+      return Responses.handleError(404, 'User with such email does not exist', res);
```

Comment:
For security reasons, I think it'd be a great idea to not tell the user if the email exist on our system or not. That way a malicious user cannot guess somebody's email. However, we can go with this, just pointing it out for future purposes.